lib/tests/streamlit/runtime/scriptrunner/script_requests_test.py

```diff
@@ -223,6 +229,22 @@ def test_on_script_yield_with_fragment_rerun_request(self):
             reqs._rerun_data, RerunData(fragment_id_queue=["my_fragment_id"])
         )
 
+    def test_on_script_yield_with_fragment_scoped_rerun(self):
```

Comment:
Is it possible to add a unit test in `fragment_test.py` where we have a fragment-scoped rerun inside of an "app" and can somehow ensure that only the fragment is rerun when triggered? I would need to look into it more closely, but I *think* we already have some similar tests where we simulate apps that we could re-use. What do you think?

Reply:
I think it'd be relatively nontrivial to simulate the fragment being part of a full "app" in a unit test. I just double-checked `fragment_test.py`, and we're decorating functions with `@st.fragment` but aren't trying to mock the rest of a script.

My hunch is that e2e test coverage will be the best way to test for this given all of the moving pieces involved that can be hard to mock (`app_session`s, `script_runner`s, etc).